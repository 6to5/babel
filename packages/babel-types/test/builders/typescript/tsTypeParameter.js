import * as t from "../../../lib/index.js";

describe("builders", function () {
  describe("typescript", function () {
    (process.env.BABEL_8_BREAKING ? describe : describe.skip)(
      "tsTypeParameter",
      function () {
        it("accept name as argument for tsTypeParameter", function () {
          const tsTypeParameter = t.tsTypeParameter(
            t.tsTypeReference(t.identifier("bar")),
            t.tsTypeReference(t.identifier("baz")),
            t.identifier("foo"),
          );
          expect(tsTypeParameter).toMatchSnapshot();
        });
        it("throws when name is missing", function () {
          expect(() => {
            t.tsTypeParameter(
              t.tsTypeReference(t.identifier("bar")),
              t.tsTypeReference(t.identifier("baz")),
            );
          }).toThrow(
            'Property name of TSTypeParameter expected node to be of a type ["Identifier"] but instead got undefined',
          );
        });
      },
    );
    (process.env.BABEL_8_BREAKING ? describe.skip : describe)(
      "tsTypeParameter - Babel 7",
      function () {
        it("accept name as argument for tsTypeParameter", function () {
          const tsTypeParameter = t.tsTypeParameter(
            t.tsTypeReference(t.identifier("bar")),
            t.tsTypeReference(t.identifier("baz")),
            "foo",
          );
          expect(tsTypeParameter).toMatchSnapshot();
        });
        it("throws when name is missing", function () {
          expect(() => {
            t.tsTypeParameter(
              t.tsTypeReference(t.identifier("bar")),
              t.tsTypeReference(t.identifier("baz")),
            );
          }).toThrow("Property name expected type of string but got undefined");
        });
      },
    );
  });
});
