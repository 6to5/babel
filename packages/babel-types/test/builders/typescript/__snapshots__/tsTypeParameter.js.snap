// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builders typescript tsTypeParameter - Babel 7 accept name as argument for tsTypeParameter 1`] = `
Object {
  "constraint": Object {
    "type": "TSTypeReference",
    "typeName": Object {
      "name": "bar",
      "type": "Identifier",
    },
    "typeParameters": null,
  },
  "default": Object {
    "type": "TSTypeReference",
    "typeName": Object {
      "name": "baz",
      "type": "Identifier",
    },
    "typeParameters": null,
  },
  "name": "foo",
  "type": "TSTypeParameter",
}
`;

exports[`builders typescript tsTypeParameter - Babel 7 accept name as argument for tsTypeParameter 2`] = `
Object {
  "constraint": Object {
    "type": "TSTypeReference",
    "typeName": Object {
      "name": "bar",
      "type": "Identifier",
    },
    "typeParameters": null,
  },
  "default": Object {
    "type": "TSTypeReference",
    "typeName": Object {
      "name": "baz",
      "type": "Identifier",
    },
    "typeParameters": null,
  },
  "name": "foo",
  "type": "TSTypeParameter",
}
`;

exports[`builders typescript tsTypeParameter accept name as argument for tsTypeParameter 1`] = `
Object {
  "constraint": Object {
    "type": "TSTypeReference",
    "typeName": Object {
      "name": "bar",
      "type": "Identifier",
    },
    "typeParameters": null,
  },
  "default": Object {
    "type": "TSTypeReference",
    "typeName": Object {
      "name": "baz",
      "type": "Identifier",
    },
    "typeParameters": null,
  },
  "name": Object {
    "name": "foo",
    "type": "Identifier",
  },
  "type": "TSTypeParameter",
}
`;

exports[`builders typescript tsTypeParameter accept name as argument for tsTypeParameter 2`] = `
Object {
  "constraint": Object {
    "type": "TSTypeReference",
    "typeName": Object {
      "name": "bar",
      "type": "Identifier",
    },
    "typeParameters": null,
  },
  "default": Object {
    "type": "TSTypeReference",
    "typeName": Object {
      "name": "baz",
      "type": "Identifier",
    },
    "typeParameters": null,
  },
  "name": Object {
    "name": "foo",
    "type": "Identifier",
  },
  "type": "TSTypeParameter",
}
`;
